AdobeStockImageAdminUi/view/adminhtml/web/js/components/grid/column/image-preview.js

```diff
@@ -50,10 +52,39 @@ define([
                 }
             ],
             listens: {
-                '${ $.sortByComponentName }:applied': 'hide'
+                '${ $.sortByComponentName }:applied': 'hide',
+                '${ $.bookmarksProvider }:activeIndex': 'onActiveIndexChange'
             }
         },
 
+        /**
+         * Listener of the activeIndex property.
+         */
+        onActiveIndexChange: function () {
```

Comment:
I'd apply this fix to magento2 `image-preview` and actually subscribe to `lastOpenedImage` instead of bookmarks index change:

in `initialize` method:

```
            this.lastOpenedImage.subscribe(function (newValue) {
                if (newValue === false && _.isNull(this.visibleRecord())) {
                    return;
                }
                if (newValue === this.visibleRecord()) {
                    return;
                }

                if (newValue === false) {
                    this.hide();
                    return;
                }

                this.show(this.masonry().rows()[newValue]);
            }.bind(this));
```

The outdated preview displayed image can be also fixed in the magento2 `image-preview` by adding a listener `'${ $.provider }:data.items': 'updateDisplayedRecord'` and updating the preview (if it is opened) contents when the data provider is updated:

```
        updateDisplayedRecord: function (items) {
            if (!_.isNull(this.visibleRecord())) {
                this.displayedRecord(items[this.visibleRecord()]);
            }
        },
```